ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java

```diff
@@ -101,19 +103,19 @@ public JsonRpcResponse response(final JsonRpcRequestContext request) {
             .map(blockHeader -> blockHeader.getBaseFee().orElse(0L))
             .orElseGet(
                 () ->
-                    protocolSchedule
-                        .getByBlockNumber(nextBlockNumber)
-                        .getEip1559()
+                    Optional.of(protocolSchedule.getByBlockNumber(nextBlockNumber).getFeeMarket())
```

Comment:
Not a fan of this lambda style.  `Optional.of.filter.map.else` is really just `var x; if (x.y) ... else ...` and I think the latter reads better.

Reply:
I can appreciate easier reading, but in these two cases I am doing it this way to avoid using explicit casts.  I have linter ptsd from prior project ;)

Reply:
Perhaps we could move `computeBaseFee(
      final long blockNumber,
      final long parentBaseFee,
      final long parentBlockGasUsed,
      final long targetGasUsed)` into the `FeeMarket` interface with a default implementation of `return 0L;`?